PORT-8538 Add general security page to Port's docs (#1424)
port-labs/port-docs: docs/security/security.md

---
title: Security & Compliance
sidebar_label: Security & Compliance
sidebar_class_name: custom-sidebar-item sidebar-menu-resources
---

# Security & Compliance

> Security and compliance are always our top priority at Port. We're building a product that's meant to be the daily hub for all of your developers, when fully integrated in an organization, Port becomes a main junction for the developers' daily routines, as well as a centerpiece for analytics, metrics, self-service and automation flows, this can only be achieved if we have your trust and confidence that we will treat your data with respect and security in mind.

Reviewer: ```suggestion
Security and compliance are always our top priority at Port. We're building a product that's meant to be the daily hub for all of your developers. When fully integrated in an organization, Port becomes a main junction for the developers' daily routines, as well as a centerpiece for analytics, metrics, self-service and automation flows. This can only be achieved if we have your trust and confidence that we will treat your data with respect and security in mind.
```

Port uses a separate virtual private network for each of its environments, and enforces complete segregation between test, staging and production environments.

### Intrusion detection

Port uses [AWS GuardDuty](https://aws.amazon.com/guardduty/) as its intrusion detection system (IDS), this system relies on both signature-based security and algorithm-based security to identify traffic patterns that are similar to known attack methods.

### Rate limiting and traffic control

Port uses [AWS WAF](https://aws.amazon.com/waf/) as its Web Application Firewall (WAF) solution, this is an additional security and control layer, added to the built-in Dos/DDoS protections that AWS provides out-of-the-box across its infrastructure.

AWS WAF protects Port's infrastructure both from well-known bot attacks, as well as defines custom rules that enforce [rate limiting](/docs/api-reference/rate-limits.mdx) and unwanted traffic control.

## Data security and privacy

### Data encryption

> All data in Port servers is encrypted at rest, data at rest is encrypted using AES-256bit encryption. AWS stores and manages data cryptography keys in its redundant and globally distributed Key Management Service ([AWS KMS](https://aws.amazon.com/kms/)). So, if an intruder were ever able to access any of the physical storage devices, the Port data contained therein would still be impossible to decrypt without the keys, rendering the information a useless jumble of random characters.

Reviewer: ```suggestion
All data in Port servers is encrypted at rest using AES-256bit encryption. AWS stores and manages data cryptography keys in its redundant and globally distributed Key Management Service ([AWS KMS](https://aws.amazon.com/kms/)). So, if an intruder were ever able to access any of the physical storage devices, the Port data contained there would still be impossible to decrypt without the keys, rendering the information a useless jumble of random characters.
```

Reviewer: `rendering the information a useless jumble of random characters.` Is this necessary?  :D

### Business Continuity

Port keeps hourly encrypted backups of data in multiple regions on AWS. While never expected, in the case of production data loss (i.e., primary data stores lost), we will restore organizational data from these backups.

## Data flow

### Data into the system

Data can be sent to Port through a variety of sources and interfaces - through its API, Ocean integrations, Git provider apps, using webhooks and more.

Data is always sent to Port using push methods, and Port never independently makes outbound requests to your infrastructure or environment to get and query information that will be ingested.

All ways of ingesting data into Port ensure that data is encrypted at rest and in transit, and any request that sends data to Port must be authenticated in order to get accepted and be processed by the system.

### Data out of system

Data ingested into Port can be accessed via Port's user interface and REST APIs.

> Port will never make an outbound request to your infrastructure or environment in order to pull information, data is always sent to Port using push methods.

Reviewer: ```suggestion
Port will never make an outbound request to your infrastructure or environment in order to pull information.
```

To trigger self-service actions and automations, Port supports a variety of backends that change how the action or automation request is sent to your infrastructure.

> Organizations that are more security concerned, limited by compliance requirements or internal policy you have the option to listen to action triggers by listening to messages coming in from a Kafka topic provided to you by Port, or you can use Port's agent, which is hosted on your premise and listens to the topic on your behalf in order to forward it to the correct destination internally. These two solutions make it possible to receive action triggers without opening any inbound network connectivity to your infrastructure.

Reviewer 2: ```suggestion
Organizations that are more security concerned, limited by compliance requirements or internal policy - you have the option to listen to action triggers by listening to messages coming in from a Kafka topic provided to you by Port, or you can use Port's agent, which is hosted on your premise and listens to the topic on your behalf in order to forward it to the correct destination internally. These two solutions make it possible to receive action triggers without opening any inbound network connectivity to your infrastructure.
```

> Aside from those, Port also supports other backends such as webhooks which make it possible for Port to directly make a request to your environment and trigger your backend, should you choose to use it.

Reviewer: I would add a short sentence regarding our custom headers which can be validated by the customer

## Applications and integrations

### Applications

> Port provides native applications for GitHub and Bitbucket which seamlessly integrate with the Git providers to ingest information and keep an up to date catalog available in Port. These integrations use OAuth 2.0 for authentication and permissions to your Git provider.

Reviewer 2: ```suggestion
Port provides native applications for GitHub and Bitbucket which seamlessly integrate with the Git providers to ingest information and keep an up-to-date catalog available in Port. These integrations use OAuth 2.0 for authentication and permissions to your Git provider.
```

> As part of the installation Port's Git provider apps, the apps gain a limited set of permissions ([GitHub](/docs/build-your-software-catalog/sync-data-to-catalog/git/github/github.md#permissions), [Bitbucket](/docs/build-your-software-catalog/sync-data-to-catalog/git/bitbucket/bitbucket.md#permissions)). In addition, it is possible to selectively authorize access to a limited set of repositories.

Reviewer 2: ```suggestion
As part of the installation of Port's Git provider apps, the apps gain a limited set of permissions ([GitHub](/docs/build-your-software-catalog/sync-data-to-catalog/git/github/github.md#permissions), [Bitbucket](/docs/build-your-software-catalog/sync-data-to-catalog/git/bitbucket/bitbucket.md#permissions)). In addition, it is possible to selectively authorize access to a limited set of repositories.
```

> By default the Git integrations will not ingest any source code, and will only search for `README.md` files specifically. It is possible to configure the apps to not ingest any source code files at all, or to ingest additional files that are relevant for your use case.

Reviewer 2: ```suggestion
By default the Git integrations will not ingest any source code, and will only search for `README.md` files specifically. It is possible to configure the apps to not ingest any source code files at all, or to ingest additional files that are relevant to your use case.
```

### Open source integrations

> Port provides a variety of open source integration for different use cases, such as the K8s exporter and Port agent. These integrations are completely open source and you have the option to view the source code on GitHub. In addition, these integrations are fully customizable and are deployed on your premise, so they do not require opening inbound network connectivity to your infrastructure or providing any secrets or credentials to Port

Reviewer 2: ```suggestion
Port provides a variety of open source integration for different use cases, such as the K8s exporter and Port agent. These integrations are completely open source and you have the option to view the source code on GitHub. In addition, these integrations are fully customizable and are deployed on your premise, so they do not require opening inbound network connectivity to your infrastructure or providing any secrets or credentials to Port.
```

Deploying and running an Ocean integration on premise ensures there is no need to open inbound network connectivity to your infrastructure or provide any secrets or credentials to Port.

Ocean integrations act as a proxy between Port and your environment, securely handling outbound connections, encrypting data during transit and deliberately controlling the access Port has to your data. Sensitive credentials stay behind your firewall.

Ocean integrations work by querying the data source (either an internal resource or a third-party service), parsing and modifying the data according to the user configuration, and then sending the resulting entries directly to Port. This means Port only receives the end result, and all data is processed on your premise. Ocean integrations come built in with PII and secret keys scrubbing and masking capabilities. Combined with the on premise data processing, this methodology gives you confidence that no secret or confidential information leaves your premise.

### "Hosted by Port" integrations

> For some third party services, a "Hosted by Port" integration option is available. In this deployment type, the integration is hosted and managed on Port's infrastructure, removing the need to personally host, manage and maintain the integration on your premise.

Reviewer 2: ```suggestion
For some third-party services, a "Hosted by Port" integration option is available. In this deployment type, the integration is hosted and managed on Port's infrastructure, removing the need to personally host, manage and maintain the integration on your premise.
```

> While this deployment method does require providing an access token to the relevant third party service in order for the integration to work, it still provides the same level of security and customizability that all Ocean integrations provide.

Reviewer 2: ```suggestion
While this deployment method does require providing an access token to the relevant third-party service in order for the integration to work, it still provides the same level of security and customizability that all Ocean integrations provide.
```

> When an integration is hosted by Port, it is deployed on Port's infrastructure and makes calls to the third party service, receives the results, processes the data according to the integration configuration, and then sends the processed results to Port. While this does mean data is processed outside of your premise, there is still complete separation between the integration and Port itself, meaning an additional layer of protection that prevents completely unprocessed data from making its way to your Port account.

Reviewer 2: ```suggestion
When an integration is hosted by Port, it is deployed on Port's infrastructure and makes calls to the third-party service, receives the results, processes the data according to the integration configuration, and then sends the processed results to Port. While this does mean data is processed outside of your premise, there is still complete separation between the integration and Port itself, meaning an additional layer of protection that prevents completely unprocessed data from making its way to your Port account.
```

## Application security

### Multi-factor authentication

In addition to password login and social login, multi-factor authentication (MFA) provides an added layer of security to Sentry. We encourage MFA as an important step towards securing data access from intruders. Port users can use time-based one-time password (TOTP) apps like Google Authenticator as additional factors. This also applies to sign-in with an SSO provider.

> Sentry’s organization list also displays who has MFA enabled so users can vet their own organization’s security.

Reviewer 2: ```suggestion
Port’s organization list also displays who has MFA enabled so users can vet their own organization’s security.
```

### Single sign-on (SSO)

> Port integrates with all major [SSO providers](/docs/sso-rbac/sso-rbac.md), allowing you to seamlessly import users, and teams, and also exposing Port to your organization in a managed manner.

Reviewer 2: ```suggestion
Port integrates with all major [SSO providers](/docs/sso-rbac/sso-rbac.md), allowing you to seamlessly import users and teams, and also exposing Port to your organization in a managed manner.
```

### REST API authentication

Port's REST API uses a [JWT auth token](/docs/build-your-software-catalog/custom-integration/api/api.md#get-api-token) for authentication. The token is generated by making an authentication request using a unique Client ID and Client Secret.

The authentication tokens are passed using the authorization header and are used to authenticate with the API.

### Vulnerability remediation

Port uses [GitHub dependabot](https://docs.github.com/en/code-security/getting-started/dependabot-quickstart-guide) and [AWS ECR](https://aws.amazon.com/ecr/) to scan, detect and remediate any vulnerabilities in the code libraries and packages used to provide the Port product. In addition, Port uses [SonarCloud](https://www.sonarsource.com/solutions/security/) for Static Application Security Testing (SAST).

When a vulnerability is detected, it is handled based on its classified severity according to the following timeline:

- Critical: up to 2 days
- High: up to 7 days
- Medium: up to 45 days
- Low: up to 90 days

### Secure application development (application development lifecycle)

> Port practices continuous delivery, which means all code changes are committed, tested, shipped, and iterated on in a rapid sequence. A continuous delivery methodology, complemented by pull request, continuous integration (CI), and automated error tracking, significantly decreases the likelihood of a security issue and improves the response time to and the effective eradication of bugs and vulnerabilities. Release notes and details for Port releases can be found in the [changelog](https://roadmap.getport.io/changelog).

Reviewer 2: ```suggestion
Port practices continuous delivery, which means all code changes are committed, tested, shipped, and iterated on in a rapid sequence. A continuous delivery methodology, complemented by pull request, continuous integration (CI), and automated error tracking, significantly decreases the likelihood of a security issue and improves the response time and the effective eradication of bugs and vulnerabilities. Release notes and details for Port releases can be found in the [changelog](https://roadmap.getport.io/changelog).
```

### Secret storage and handling

Port does not require sharing and storing any secrets with it. Port does allow for secret storage in order to improve ease of use and enable specific features and use cases, however those features can always be implemented in alternative ways that do not require sharing secrets with Port.

> Any and all secrets shared with Port are encrypted both at rest and in transit, once they are saved into the system, their value cannot be retrieved directly, it will only be used and referenced as specified by the user. A customer's secrets are completely isolated from another's at both the data storage layer and the access permissions layer.

Reviewer 2: ```suggestion
Any and all secrets shared with Port are encrypted both at rest and in transit, Once they are saved into the system, their value cannot be retrieved directly, it will only be used and referenced as specified by the user. A customer's secrets are completely isolated from another's at both the data storage layer and the access permissions layer.
```

### Security policies

Port maintains an internal wiki of security policies, which is updated on an ongoing basis and reviewed annually for gaps. An overview of specific security policies is available to Port enterprise customers upon request:

- Access Management
- Backup Management
- Change Management
- Data Retention Management
- Information Security
- Incident Response
- Risk Management
- Vulnerability Management

### Security training

All new employees receive onboarding and systems training, including environment and permissions setup, formal software development training (if pertinent), security policies review, company policies review, and corporate values and ethics training.

All employees additionally complete security training at least once a year. Policies presented to employees as part of the onboarding process are reviewed once a year to ensure we are keeping up with best practices.

### Disclosure Policy

> Port notifies customers of any data breaches as soon as possible via email and the primary contact channel, followed by multiple periodic updates throughout each day addressing progress and impact. Port Enterprise plans include a dedicated customer success manager who holds responsibility for customer communication, as well as regular check-ins and escalations.

Reviewer 2: ```suggestion
Port will notify customers of any data breaches as soon as possible via email and the primary contact channel, followed by multiple periodic updates throughout each day addressing progress and impact. Port Enterprise plans include a dedicated customer success manager who holds responsibility for customer communication, as well as regular check-ins and escalations.
```

## Certifications and compliance

Port has obtained the following compliance certifications:

- SOC2 Type II
- ISO 27001:2022
- GDPR

It is always possible to receive a copy of these reports and certificates by contacting either info@getport.io or your Port account manager.

## Penetration testing

Port undergoes annual penetration testing conducted by an independent, third-party agency. As part of the test, Port provides the agency with an isolated environment of Port and a high-level diagram of the application architecture. No customer data is exposed to the agency through penetration testing.

Information about any security vulnerabilities successfully exploited through penetration testing is used to set mitigation and remediation priorities. A summary of penetration test findings is available to customers upon request.

# Privacy policy

> You can find Port's privacy policy [here](https://www.getport.io/legal/privacy-policy)

Reviewer 2: ```suggestion
You can find Port's privacy policy [here](https://www.getport.io/legal/privacy-policy).
```

## Data protection addendum

> You can find Port's DPA [here](https://www.getport.io/legal/dpa)

Reviewer 2: ```suggestion
You can find Port's DPA [here](https://www.getport.io/legal/dpa).
```

---
title: Security & Compliance
sidebar_label: Security & Compliance
sidebar_class_name: custom-sidebar-item sidebar-menu-resources
---

# Security & Compliance

> Port is an internal developer portal platform and as such we place the utmost importance on data security, privacy and compliance, so that you can trust that your compliance, privacy and security needs are met.

Reviewer 3: ```suggestion
Port is an internal developer portal platform and as such we place the utmost importance on data security, privacy and compliance, so that you can trust that your security needs are met.
```

> The goal of this document is to provide you with transparency as to security policies and practices at Port. 

Reviewer 3: ```suggestion
The goal of this document is to provide you with transparency as to Port's security policies and practices. 
```

Port uses a variety of industry-standard technologies and services to secure your data from unauthorized access, disclosure, use, and loss. All Port employees are trained on security practices during company onboarding and on an annual basis.

Security is managed by Port’s Chief Technology Officer and maintained by Port’s Security & DevOps team.

> This document will outline the different measures, protocols, processes and methods we take to ensure that your data stays secure and highly available.

Reviewer 3: ```suggestion
This document will outline the different measures, protocols, processes and methods we use to ensure that your data stays secure and highly available.
```
